File: Larchik.Domain/Account.cs
namespace Larchik.Domain;

public class Account
{
    public Guid Id { get; set; }
    public AppUser User { get; set; } = null!;
    public string UserId { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public ICollection<Asset> Assets { get; set; } = new List<Asset>();
    public ICollection<Deal> Deals { get; set; } = new List<Deal>();
}

File: client-app/src/app/models/account.ts
import { Asset } from "./asset";

export interface Account {
    id: string;
    name: string;
    assets: Asset[];
}

export class Account implements Account {
    constructor(init?: AccountFormValues) {
      Object.assign(this, init);
    }
  }

export class AccountFormValues {
    id?: string = undefined;
    name: string = '';

    constructor(account?: AccountFormValues) {
        if (account) {
            this.id = account.id;
            this.name = account.name;
        }
    }
}